src/users/templates/user_profile.html

```diff
@@ -72,7 +72,7 @@
             </div>
             {% if user == request.user %}
             <div id="edit-profile">
-                <a href="{% url 'update_profile' user.pk %}">
+                <a href="{% url 'update_profile' request.user.userprofile.pk %}">
```

Comment:
I would change it to `pk=request.user.userprofile.pk` to be more explicit. That way you will know what the parameter is in the url/view you are passing the value to.